src/profile.js

```diff
@@ -25,46 +25,53 @@ import {withRouter} from 'react-router-dom'
 
     handleUpdate = (event) => {
       event.preventDefault();
-      if (this.checkAge()) {
-        if (this.completedInput()){
-
-          const id = this.props?.location?.state?.id;
-          console.log(id);
-          const myForm = new FormData (document.getElementById("profileForm"));
-          myForm.append("id", id);
-          const myRequest = new Request("http://127.0.0.1:5000/profile", {
-            method: "POST",
-            body: myForm,
-          });
-
-          fetch(myRequest)
-          .then((res) =>
+      if (this.completedInput()){
+        if (this.checkValideDate()) {
+          if (this.checkAge() > 18 && this.checkAge() < 100){
```

Comment:
maybe save this.checkAge() as a variable so you don't have to call the function twice?